src/nanoarrow/common/array.c

```diff
@@ -720,7 +731,40 @@ static int ArrowArrayViewSetArrayInternal(struct ArrowArrayView* array_view,
     }
   }
 
-  // Check the number of buffers
+  if (array_view->storage_type == NANOARROW_TYPE_STRING_VIEW ||
+      array_view->storage_type == NANOARROW_TYPE_BINARY_VIEW) {
+    const int64_t n_buffers = array->n_buffers;
+    const int64_t n_fixed_buffers = 2;
+
+    int64_t n_variadic_buffers = n_buffers - n_fixed_buffers - 1;
+    // Theoretically if the variadic buffers are not used the above subtraction
```

Comment:
Not sure if this is true - need to add a test when there is an empty StringView array